### PR TITLE
DHCP address tolerance

### PR DESCRIPTION
**Pull request by hickeng**
This adds support for:
* structured renaming of NICs
* publishing of DHCP addresses on interfaces
* use of DHCP address if already assigned to interface

This is broken out from the larger change for #746

### CODE REVIEW

File: lib/tether/ops_linux.go [hunk 1]
```diff
@@ -143,40 +143,249 @@ func linkBySlot(slot int32) (netlink.Link, error) {
...
+		log.Infof("Added hosts entry: %s", entry)
 	}
 
-	// TODO update /etc/hosts and /etc/resolv.conf
+	// Add nameservers
+	// This is incredibly trivial for now - should be updated to a less messy approach
+	if len(endpoint.Network.Nameservers) > 0 {
+		resolv, err := os.OpenFile(resolvFile, os.O_APPEND|os.O_WRONLY, 0644)
+		if err != nil {
+			detail := fmt.Sprintf("failed to update resolv.confg for endpoint %s: %s", endpoint.Network.Name, err)
```
Comment (dougm) [inline lib/tether/ops_linux.go]: typo 'resolv.confg' - could just add another %s and resolvFile param.

**Comment by dougm** [comment]
lgtm

**Comment by andrewtchin** [comment]
lgtm

**Comment by hickeng** [comment]
LGTM

#why don't the prior two count?

**Pull request merged**